controlplane/controllers/ck8scontrolplane_controller.go

```diff
@@ -699,12 +699,7 @@ func (r *CK8sControlPlaneReconciler) syncMachines(ctx context.Context, kcp *cont
 
 		patchHelper, err := patch.NewHelper(m, r.Client)
 		if err != nil {
-			return fmt.Errorf("failed to create patch helper for machine: %w", err)
-		}
-
-		// Create a new map if machine has no annotations.
-		if m.Annotations == nil {
-			m.Annotations = map[string]string{}
+			return err
```

Comment:
This will panic if no annotation is set. 

a) Use `m.GetAnnotations()` to avoid that
b) Wrap the error you return here